tasks/scrapling_research/src_parser.py

```
@@ -0,0 +1,1363 @@
+                value = value.replace('"', r"\"")  # Escape double quotes in user input
+                # Not escaping anything with the key so the user can pass patterns like {'href*': '/p/'} or get errors :)
+                selector += '[{}="{}"]'.format(key, value)
+            if selector != "*":
+                selectors.append(selector)
+
+        if selectors:
+            results = cast(Selectors, self.css(", ".join(selectors)))
+            if results:
+                # From the results, get the ones that fulfill passed regex patterns
+                for pattern in patterns:
+                    results = results.filter(lambda e: e.text.re(pattern, check_match=True))
```

Ruff (0.15.2)

[warning] Function definition does not bind loop variable `pattern`

(B023)